src/client.js

```diff
@@ -136,14 +137,27 @@ export default class StitchClient {
    */
   authenticate(providerType, options = {}) {
     // reuse existing auth if present
-    return this.auth.getAccessToken().then(accessToken => {
+    return Promise.all(
+      [
+        this.auth.getAccessToken(),
+        this.auth.getLoggedInProviderType()
+      ]
+    ).then(([accessToken, loggedInProviderType]) => {
       if (accessToken) {
-        return this.auth.authedId();
+        if (providerType === PROVIDER_TYPE_ANON && loggedInProviderType === PROVIDER_TYPE_ANON) {
+          return false; // is authenticated, skip log in
+        }
+
+        return this.logout(); // will not be authenticated, continue log in
```

Comment:
[minor] I think logout we really don't care if it fails or not, just that we try and wait to do it. Given my comment on the logout method holds, we can just do `this.logout().then(() => true)`

Reply:
good point. will do